### DISCUSSION TITLE
Will we support Laravel 6 or not?

### DISCUSSION THREAD

**Discussion by andrey-helldar** [Laravel-Lang/common#18] [Polls]
@Laravel-Lang/laravel-lang, hi!

With the new project structure, we can easily translate any files of any version of any package.

According to our compatibility list, we are working with Laravel 7-9, but Laravel 6-7 [`out of support`](https://laravel.com/docs/9.x/releases#support-policy) by Laravel team.

During development, I automatically added support for Laravel 6.

**Comment by ariaieboy** [comment]
according to the [packagist stats](https://packagist.org/packages/laravel/framework/stats), laravel 6 have more download than laravel 7 and since its LTS version, it will receive security update for more than 2 months till 6 September if adding support for laravel 6 will not add more work overhead we can add support for laravel 6 too.
another good approach is to only support versions that are supported by laravel that is versions 8 and 9

**Comment by luisprmat** [comment]
Laravel 6 was a LTS version and it has security fixes until september 6 of 2022, i think we must support versions until its end of life.

![Screenshot_20220630_061355](https://user-images.githubusercontent.com/9275870/176664353-23edfaca-1d27-42cb-8412-00df53af50a8.jpg)

**Reply by andrey-helldar** [reply]
On the other hand, there will be updates for critical vulnerabilities until September, despite the fact that the main development of this version has been stopped.

In addition, support for Laravel 7 has also been discontinued.

**Reply by andrey-helldar** [reply]
@Laravel-Lang/laravel-lang, therefore, I would even raise the question of whether it is worth serving translations for Laravel 7...

**Comment by andrey-helldar** [comment]
At the same time, it is worth considering that when updating the [`publisher`](https://publisher.laravel-lang.com), the minimum PHP version is 8.1, since I used 8.1 features.

Unfortunately Lararavel 7 is not compatible with PHP 8.1 - it works with PHP 8.0 at most. Therefore, the publisher can only be installed with Laravel 8 and 9.

But this does not affect this package in any way, since it is not tied to our publisher. And its files can be used both in raw form and in [third-party publishers](https://laravel-lang.com/installation/managers.html)..

**Comment by ariaieboy** [comment]
@luisprmat since laravel 6 only receives security fixes for 68 days it's not nesseccary to support it in our latest version. 

and if i am not wrong our prev version is not laravel 6 compatible too. 

@andrey-helldar i think for this version we should drop laravel 7 support instead of adding another older version support.

**Reply by andrey-helldar** [reply]
I agree with @ariaieboy. Given that the latest version of the project that supports Laravel 6 is 5.0, and today we have released the 11th version.

**Comment by andrey-helldar** [comment]
@Laravel-Lang/laravel-lang, removing files from a project is easy, so I'll wait to see what the team poll shows ☝️

**Reply by andrey-helldar** [reply]
So, I came back to this survey and I have the following thoughts on this subject:

1. I propose to support translation versions for projects only at the stage of their active life. I propose not to consider the period of receiving security bug fixes as an active period, since this is more suitable for the description of post-mortem maintenance. For example, Laravel 8 has its last active day on July 26, 2022.

2. I will wait until September 3rd as the day the purge starts and remove support for old versions of projects from the project. These include the following projects and their versions:

    - Laravel Framework: 6.x, 7.x, 8.x
    - Lumen Framework: 6.x, 7.x, 8.x
    - Laravel UI: 1.x, 2.x

3. I suggest supporting older versions of minor packages as long as they are compatible with active versions of Laravel. For example, the following older packages support Laravel 9:

    - Jetstream: 1.x
    - Nova: 3.x
    - Laravel UI: 3.x

@Laravel-Lang/laravel-lang, what do you think?

**Reply by mohamedsabil83** [reply]
Agree

**Reply by dinhquochan** [reply]
Agree absolutely!

**Reply by luisprmat** [reply]
I agree

**Reply by EuCarlos** [reply]
I agree too

**Reply by andrey-helldar** [reply]
For convenience, I have updated the compatibility page: https://laravel-lang.com/installation/compatibility.html

**Reply by makowskid** [reply]
6.0 is LTS, so unless there's a new LTS version announced, we should be supporting it

**Reply by andrey-helldar** [reply]
@makowskid, Taylor stops releasing security fixes for version 6 on September 3, 2022.

LTS means that this version will be supported longer than others, and that's exactly what happened. Look at the dates on the official timetable: https://laravel.com/docs/9.x/releases#support-policy

Maintenance of version 7 ended in October 2020, and version 6 ended in January 2022. Starting in January 2022, version 6 does not receive scheduled updates, fixes, or improvements. Only security fixes. It is fashionable to compare with post-mortem service. That's what LTS means.

This is why I suggested using September 3rd as the final death date for Laravel 6.

**Reply by makowskid** [reply]
I've only mentioned this because L6 was LTS for so long. I'm not sure what's the current situation but L9 was supposed to be a new LTS. So is it or not finally? :-)
I'll leave it up to you guys - I think we should support versions that are officially supported as well as a minimum. I've dealt with simple packages with quite old Laravel versions support, there we no issues usually. I think this package is also simple enough to keep much older versions up and running :)

**Reply by andrey-helldar** [reply]
Laravel 6 will finally die on September 3rd. No new LTS has been announced. Official information can be found here: https://laravel.com/docs/9.x/releases#support-policy

In our case, the old packages are already translated and there is no need to maintain them. If you need to use old translations, just install the package of the boring version: https://laravel-lang.com/installation/compatibility.html

At the same time, our publisher works with PHP 8.1 and higher. This version is supported by Laravel 8 and 9. This means that the latest version cannot be installed, for example, on Laravel 7. Therefore, you will also have to use older versions. And that's okay.

https://publisher.laravel-lang.com

**Reply by andrey-helldar** [reply]
After the end of support date, there is a secure fixes period. This means that during this period, the application version will receive only security updates. Correcting grammatical errors and, moreover, adding new phrases are not security errors. Therefore, it makes no sense to support these versions, since they will obviously be translated by us by that time.

As a last resort, as I wrote above, you can find the required version in the [compatibility table](https://laravel-lang.com/installation/compatibility.html) and install it.

![image](https://user-images.githubusercontent.com/10347617/184535972-5f89538d-abaa-4891-b8c3-9a2a1dcddb2a.png)

According to this table, the period of active life of versions:

- Laravel 6: September 3rd, 2019 - January 25th, 2022
- Laravel 7: March 3rd, 2020 - October 6th, 2020
- Laravel 8: September 8th, 2020 - July 26th, 2022
- Laravel 9: February 8th, 2022 - August 8th, 2023
- Laravel 10: February 7th, 2023 - August 7th, 2024

For all Laravel releases, bug fixes are provided for 18 months and security fixes are provided for 2 years (24 months).
